### PR TITLE
Consumer k8s client support

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: pkg/k8s_utils/k8s_client.go [hunk 1]
```diff
@@ -33,65 +33,57 @@ const (
...
 		return "", utils.Error{Description: utils.ErrFetchingNamespace, BackendError: "namespace empty"}
 	}
 
 	return namespace, nil
 }
+
+// ValidateK8SClient ...
+func ValidateK8SClient(k8sClient *KubernetesClient) error {
+	if k8sClient == nil {
+		return utils.Error{Description: "K8S client is nil"}
```
Comment (GunaKKIBM) [inline pkg/k8s_utils/k8s_client.go:85]: Move this to secret-common-lib

**Review by skorati — APPROVED** [review]
looks good.

**Review by prankulmahajan — APPROVED** [review]
Looks good Guna. Minor suggestion, can you update the client code as well?

File: pkg/k8s_utils/fake_k8s_client.go [hunk 2]
```diff
@@ -53,21 +51,18 @@ func FakeCreateSecret(kc KubernetesClient, fakeAuthType, secretdatafilepath stri
...
 	if err != nil {
-		kc.logger.Error("Error reading secret data", zap.Error(err))
 		return err
 	}
 
 	data[dataname] = byteData
 	secret.Data = data
-	clientset := kc.clientset
-	_, err = clientset.CoreV1().Secrets("kube-system").Create(context.TODO(), secret, metav1.CreateOptions{})
+	_, err = kc.Clientset.CoreV1().Secrets("kube-system").Create(context.TODO(), secret, metav1.CreateOptions{})
```
Comment (ambiknai) [inline pkg/k8s_utils/fake_k8s_client.go:64]: Should this be in a constant var?  => `kube-system`

Comment (GunaKKIBM) [reply pkg/k8s_utils/fake_k8s_client.go:64]: Will change this kc.Namespace

**Review by ambiknai — APPROVED** [review]

**Pull request merged**